OpenRA.Platforms.Default/Shader.cs

```diff
@@ -189,14 +190,24 @@ public void SetTexture(string name, ITexture t)
 				textures[texUnit] = t;
 		}
 
+		int GetLocation(string name)
+		{
+			if (!uniformCache.TryGetValue(name, out var param))
+			{
+				param = OpenGL.glGetUniformLocation(program, name);
```

Comment:
This could be done at the end of the ctor, and then the dictionary value used directly.

Reply:
Ah, we already have a system for retreating all uniforms. I had to travel so I just made a quick PR without checking